### PR TITLE
Introduce compose-extras.yml for custom mounting of volumes

### PR DESCRIPTION
**Pull request by oskosk**
Introduces ability to easily declare custom mounts of directories into the container.

#### Changes proposed in this Pull Request:

* Adds `docker/compose-extras.yml.sample` with sample lines for mounting a local dir into the container when it's started by `yarn docker:up`.
* Adds the package.json script `yarn docker:compose-extras` which will copy the sample file into a gitignored file `docker/compose-extras.yml`.
* Makes the command `yarn docker:compose` always load `docker/compose-extras.yml` for taking in account the customizations.
* Adds `docker/compose-extras.yml` to .gitignore so one can modify that file freely while coding without having its changes tracked

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

New feature but not user-facing.

#### Testing instructions:

1. Checkout this branch
2. Run `yarn docker:up` (hopefully having run `yarn docker:stop` beforehand.
3. Confirm that you have a new file `docker/compose-extras.yml`.
4. Run `yarn docker:stop`.
5. Update `docker/compose-extras.yml` to include the mounting of one plugin you actually have locally somewhere else. Save and close the file
6. Run `yarn docker:up`.
7. Confirm the directory was mounted  by using `yarn docker:sh` and checking if the mount point contains the content you expected to be mounted.

#### Proposed changelog entry for your changes:

* None needed

### CODE REVIEW

**Comment by oskosk** [comment]
Thanks for the review @jeherve . Updated those docs in bdf54dfe8 now

**Comment by oskosk** [comment]
Thanks, merging!

**Comment by jeherve** [comment]
r207606-wpcom

File: docker/README.md [hunk 1]
```diff
@@ -74,6 +74,12 @@ You can set the following variables on a per-command basis (`PORT_WORDPRESS=8000
 Configurable settings are documented in the [`./docker/default.env` file](https://github.com/Automattic/jetpack/blob/master/docker/default.env).
 Customizations should go into a `./docker/.env` file you create, though, not in the `./docker/default.env` file.
 
+### Mounting extra directories into the container
```
Comment (simison) [inline docker/README.md:77]: There's a section "Custom plugins & themes in the container" which does basically the same. Let's combine those into one for clarity 💪

Comment (oskosk) [reply docker/README.md:77]: It's already done in bdf54dfe849fc940823c02e2522381754f6142e5. Content was combined. The difference is that there was no providead means of letting the extra config file live with untracked modifications inside the jetpack directory. Now, `yarn docker:up` will create the file from a template and load it for you without having to run the command with extra options in the command line

Comment (oskosk) [reply docker/README.md:77]: Basically this branchI should have pushed it years ago when we were discusing the extra `-f file.yml` file thing over DM when the instructions were laid out in the README.

**Pull request merged**